Rubberduck.Core/UI/CodeExplorer/Commands/CodeExplorerExtractInterfaceCommand.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Rubberduck.Navigation.CodeExplorer;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Refactorings.ExtractInterface;
+using Rubberduck.UI.Command.Refactorings.Notifiers;
+using Rubberduck.VBEditor.Events;
+using System;
+using System.Collections.Generic;
+
+namespace Rubberduck.UI.CodeExplorer.Commands
+{
+    public class CodeExplorerExtractInterfaceCommand : CodeExplorerCommandBase
+    {
+        private static readonly Type[] ApplicableNodes =
+        {
+            typeof(CodeExplorerComponentViewModel)
+        };
+
+        private readonly IParserStatusProvider _state;
+        private readonly ExtractInterfaceRefactoring _refactoring;
+        private readonly ExtractInterfaceFailedNotifier _failureNotifier;
+
+        public CodeExplorerExtractInterfaceCommand(
+            ExtractInterfaceRefactoring refactoring,
+            IParserStatusProvider state,
+            ExtractInterfaceFailedNotifier failureNotifier,
+            IVbeEvents vbeEvents) 
+            : base(vbeEvents)
+        {
+            _state = state;
+            _refactoring = refactoring;
+            _failureNotifier = failureNotifier;
+            AddToCanExecuteEvaluation(SpecialEvaluateCanExecute);
+        }
+
+        public sealed override IEnumerable<Type> ApplicableNodeTypes => ApplicableNodes;
+
+        private bool SpecialEvaluateCanExecute(object parameter)
+        {
+            return _state.Status == ParserState.Ready &&
+                   parameter is CodeExplorerComponentViewModel node &&
+                   ExtractInterfaceRefactoring.CanExecute((RubberduckParserState)_state, node.QualifiedSelection.Value.QualifiedName);
```

Comment:
it might be more appropriate to call `CanExecute` off of the injected instance of the refactoring as `_refactoring.CanExecute([..]);`.

That same consideration would apply to the `RefactorExtractInterfaceCommand`